### ISSUE TITLE
Relative reference in external jar is resolved as a URL instead of RELATIVE

### ISSUE THREAD

**Issue by douglasbgray** [swagger-api/swagger-parser#657]
I have a service project in which OpenAPI schemas are included via a jar file. Inside this jar file, the schema contains a relative reference to another fragment in the same jar.

These are the references:
```
api.json (service project)
 -> Persons.json (jar)
   -> Person.json (relative ref in same jar) 
```

The reference is resolvable from the service to Persons.json in the jar. However, the relative reference within the jar from Persons to Person is resolved as URL instead of RELATIVE. This generates this error:

`java.net.MalformedURLException: no protocol: /Person.json`

Investigation into `ExternalRefProcessor` shows that this code strips the leading `dot` off the reference from Persons to Person: 

https://github.com/swagger-api/swagger-parser/blob/v.1.0.33/modules/swagger-parser/src/main/java/io/swagger/parser/processors/ExternalRefProcessor.java#L163

Which causes it to be treated as a URL:

https://github.com/swagger-api/swagger-parser/blob/v.1.0.33/modules/swagger-parser/src/main/java/io/swagger/parser/processors/ExternalRefProcessor.java#L131-L134

**Simple project with steps to reproduce**: https://github.com/douglasbgray/openapi-service

**Comment by gracekarina** [comment]
Hi @douglasbgray, thanks for reporting this issue, I made some changes in this PR #658, try to test the changes locally, and it seems to do the resolving and generates the `Person` class, although the build in `openapi-service` it's still broken but for another reason. Please check it out, and let us know if that solves the issue so I can merge 👍

**Comment by douglasbgray** [comment]
Hi @gracekarina, I checked it out and the fix in PR https://github.com/swagger-api/swagger-parser/pull/658 has resolved my issue. I have updated `openapi-service` and it builds correctly using the 1.0.35-snapshot version. Thanks for the quick fix!

**Comment by gracekarina** [comment]
great! I'll do the merge.

**Comment by douglasbgray** [comment]
@georgekankava I am not sure of the protocol for closing an issue. Is this something I should do, or will you take care of it?

**Comment by gracekarina** [comment]
I'll close it no problem. It was fixed by #658
